### PR TITLE
Implemented client robot data upload

### PR DESCRIPTION
**Pull request by BrandonPacewic**
### Description
Start of server / client communication. This is the stepping stone to be able to create a menu to select your robot when entering multiplayer match mode.

### Objectives
- [x] ~~Implement data exchange protocols as outlined in [`multiplayer-design.md`](https://github.com/Autodesk/synthesis/blob/master/docs/multiplayer-design.md).~~
- [x] Finish implementing data exchange protocols as outlined in [`multiplayer-design.md`](https://github.com/Autodesk/synthesis/blob/master/docs/multiplayer-design.md).
- [x] ~~Create new server testing mode that runs through these protocols.~~
- [x] Add new option for server testing mode.

### Testing Done
- Ran communication protocol testing mode.

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1561)

### CODE REVIEW

**Comment by BrandonPacewic** [comment]
@KyroVibe Would I be correct in saying that this needs to be changed to `while (_isAlive)`?

https://github.com/Autodesk/synthesis/blob/0bf4d16fb200940054d52bb528230074e20a3deb/api/Api/Aether/Lobby/LobbyServer.cs#L88-L91

**Comment by KyroVibe** [comment]
> @KyroVibe Would I be correct in saying that this needs to be changed to `while (_isAlive)`?
> 
> https://github.com/Autodesk/synthesis/blob/0bf4d16fb200940054d52bb528230074e20a3deb/api/Api/Aether/Lobby/LobbyServer.cs#L88-L91

Yes, it was a nullable at some point but now I think it's fine

**Comment by KyroVibe** [comment]
@BrandonPacewic I have a ton of changes to fix the issues introduced with the ModeManager. I'll submit them real quick before we both work on our tasks

**Comment by KyroVibe** [comment]
> @BrandonPacewic I have a ton of changes to fix the issues introduced with the ModeManager. I'll submit them real quick before we both work on our tasks

I should be done with them in like 20 minutes at most

**Comment by BrandonPacewic** [comment]
@KyroVibe Was this supposed to already be implemented? 
https://github.com/Autodesk/synthesis/blob/77df0725d0450ee7dbfc90bbf1ac018bfb028c21/api/Api/Aether/Lobby/LobbyClient.cs#L164-L166

**Comment by KyroVibe** [comment]
> @KyroVibe Was this supposed to already be implemented? 
> https://github.com/Autodesk/synthesis/blob/77df0725d0450ee7dbfc90bbf1ac018bfb028c21/api/Api/Aether/Lobby/LobbyClient.cs#L164-L166

No that's Julian's task

**Comment by BrandonPacewic** [comment]
@KyroVibe Am I correct in saying that we now have everything except for `Host Commands` implemented?

File: api/Api/Aether/Lobby/LobbyClient.cs [hunk 1]
```diff
@@ -143,6 +144,38 @@ private class Inner : IDisposable {
...
+                var task = new Task<Result<LobbyMessage?, Exception>>(() => {
+                    var response = HandleResponseBoilerplate(new LobbyMessage { ToDataRobot = request });
+                    if (response.isError) {
+                        return response;
+                    }
+
+                    var msg = response.GetResult()!;
+                    switch (msg.MessageTypeCase) {
+                        case LobbyMessage.MessageTypeOneofCase.FromDataRobot:
+                            // TODO: Accept needed robot data from server.
```
Comment (BrandonPacewic) [inline api/Api/Aether/Lobby/LobbyClient.cs:165]: This will be done as part of [JIRA Issue - 1516](https://jira.autodesk.com/browse/AARD-1516).

**Comment by BrandonPacewic** [comment]
Moving everything into #883.

**Pull request closed without merging**